feat(editor): Make the left sidebar in Expressions editor draggable

dana-gill opened the pull request:
## Summary

This PR introduces a feature request in the Expressions editor to allow for the left sidebar to be resized for better readability

## Related Linear tickets, Github issues, and Community forum posts

- https://linear.app/n8n/issue/NODE-1924/expression-modal-allow-changing-width-of-sections
- https://community.n8n.io/t/expression-editor-pls-allow-changing-width-of-sections/56535

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

Commit:
refactor(design-system): Move resize types to a separate file

This will prevent long import statements in later commits.

Commit:
feat(editor-ui): Make the left sidebar in Expressions editor draggable

Commit:
Use camelCase instead of kebab-case for instances of N8nResizeWrapper

This encourages more type safety.

Commit:
Use throttle instead of debounce for resizing in NDVDraggablePanels

This makes it more uniformed with the Expressions editor. In addition,
throttle is the more accurate since we want resize to be called during a
certain period rather than delaying the call to resize. This also allows
us to save calls to resizing.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=11838) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=11838) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=11838) it.</sub>

Commit:
Reduce spacing for ExpressionEditModal containers

This will make the draggable cursor more visible when it is in the
middle of two Expression Editor panels.

ShireenMissi reviewed (approved):
LGTM 🎉

janober commented:
Got released with [`n8n@1.70.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.70.0)
